src/EventCalendar.js

```diff
@@ -65,120 +104,526 @@ class EventCalendar {
+    }
+
+    /**
+     * Restore all dates to their initial state (when the calendar was initialised)
+     */
+    styleClearAll (target) {
+        let _self = this,
+            $elems = _self.clndr.element.find('.day-contents');
+
+            $elems.each(function() {
+                let $elem = $(this),
+                    $elemParent = $elem.parent(),
+                    date = moment($elem.data('day')).format(_self.dateFormatLong),
+                    ariaLabel = 'Unselected';
+
+                // If event was passed in the `events` array, it will be reset back to `.selected` and needs labelling
+                if ($elemParent.attr('class').indexOf('event') >= 0) {
+                    ariaLabel = 'Selected. Event will repeat on this day';
+                }
+
+                $elem.attr('aria-label',  date + '. ' + ariaLabel);
+                $elemParent.removeClass('event-add event-del event-repeat');
+            });
+        }
+
+    /**
+     * Check whether `date` is in the same month as `dateToCompareTo`
+     * 
+     * @param {moment} dateToCompareTo Haystack
+     * @param {moment} date Needle
+     */
     static isInMonth (dateToCompareTo, date) {
-        return date.date.date.month() === dateToCompareTo.month();
+        return date.month() === dateToCompareTo.month();
     }
     
+    /**
+     * Check whether `date` is in the same year as `dateToCompareTo`
+     * 
+     * @param {moment} dateToCompareTo Haystack
+     * @param {moment} date Needle
+     */
     static isInYear (dateToCompareTo, date) {
-        return date.date.date.year() === dateToCompareTo.year();
+        return date.year() === dateToCompareTo.year();
+    }
+
+    /**
+     * Check whether two moment instances are the same date
+     * 
+     * @param {moment} a Haystack
+     * @param {moment} b Needle
+     */
+    static matchDates (a, b) {
+        return a._i != b._i;
+    }
+    
+    /**
+     * Updates the review panel to detail the number of exceptions being applied outside of any recurrence pattern. Controls
+     * the visibility of certain elements and updates text counters.
+     * 
+     * Triggered by the paintMonth() method.
+     */
+    updateReview () {
+        let _self = this,
+            numDatesToAdd = _self.clndr.options.extras.datesToAdd.length,
+            numDatesToDel = _self.clndr.options.extras.datesToDel.length,
+            datesToAddContainer = _self.clndr.element.find('.js-dates-to-add'),
+            datesToDelContainer = _self.clndr.element.find('.js-dates-to-del'),
+            resetButton = _self.clndr.element.find('.js-calendar-reset');
+
+        // Dates being added (outside of any recurrence pattern)
+        if (numDatesToAdd > 0) {
+            datesToAddContainer.html((numDatesToAdd === 1) 
+                ? numDatesToAdd + ' day will be added' : numDatesToAdd + ' days will be added').show();
```

Comment:
We could use some string interpolation here to make this expression a tad shorter:

```
datesToAddContainer.html(`day${numDatesToAdd > 1 ? 's' : ''} will be added`).show();
```

Reply:
As we're using this a few times we could add a little helper method on the class (it could also be static) that would handle this for us in one place:

```
function pluralizer9000 (noun, quantity) {
  return quantity > 1 ? noun + 's' : noun;
}

console.log(`${pluralizer9000('day', numDatesToAdd)} will be added`);
```